nautobot/docs/development/jobs/job-structure.md

```diff
@@ -0,0 +1,733 @@
+
+### `soft_time_limit`
+
+An int or float value, in seconds, which can be used to override the default [soft time limit](../../user-guide/administration/configuration/settings.md#celery_task_soft_time_limit) for a Job task to complete.
+
+The `celery.exceptions.SoftTimeLimitExceeded` exception will be raised when this soft time limit is exceeded. The Job task can catch this to clean up before the [hard time limit](../../user-guide/administration/configuration/settings.md#celery_task_time_limit) (10 minutes by default) is reached:
+
+```python
+from celery.exceptions import SoftTimeLimitExceeded
+from nautobot.apps.jobs import Job
+
+class ExampleJobWithSoftTimeLimit(Job):
+    class Meta:
+        name = "Soft Time Limit"
+        description = "Set a soft time limit of 10 seconds"
+        soft_time_limit = 10
+
+    def run(self):
+        try:
+            # code which might take longer than 10 seconds to run
+            job_code()
+        except SoftTimeLimitExceeded:
+            # any clean up code
+            cleanup_in_a_hurry()
+```
+
+### `task_queues`
+
+Default: `[]`
+
+A list of Job Queue names that the Job can be routed to. An empty list will default to only allowing the user to select the [default Celery queue](../../user-guide/administration/configuration/settings.md#celery_task_default_queue) (`default` unless changed by an administrator). The queue specified in the Job's `default_job_queue` will be used if a queue is not specified in a Job run API call.
+
++/- 2.4.0 "Changed default queue selection"
+    As a result of the addition of Job Queues, the default queue when running a Job without explicitly selecting a queue is now the job queue specified in the `default_job_queue` field on the Job model. `default_job_queue` fields for any existing Job instances are automatically populated with the name of the first entry in `task_queues` list of the Job class. When `task_queues` list on the Job class is empty, the corresponding Job instance's `default_job_queue` will be the job queue with the name provided by `settings.CELERY_TASK_DEFAULT_QUEUE`. You can also override the initial `default_job_queue` by setting `default_job_queue_override` to True and assign the field with a different Job Queue instance.
+
+!!! note
+    A worker must be listening on the requested queue or the Job will not run. See the documentation on [task queues](../../user-guide/administration/guides/celery-queues.md) for more information.
+
+### `template_name`
+
+A path relative to the Job source code containing a Django template which provides additional code to customize the Job's submission form. This template should extend the existing Job template, `extras/job.html`, otherwise the base form and functionality may not be available.
```

Comment:
The template file must be discoverable by Django's template engine, so perhaps:

```suggestion
A Django template path which provides additional code to customize the Job's submission form. This template should extend the existing Job template, `extras/job.html`, otherwise the base form and functionality may not be available.
```

Because the template must be discoverable by Django's template engine, typically the file must be provided by Nautobot itself or an App - there's no pattern presently for `JOBS_ROOT` or a Git repository to provide such templates.